### PR TITLE
MPCTensor : Numpy Hook Methods

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Implementation of Numpy Methods on MPCTensor which performs it on the underlying tensor

Implemented Ops:
- [x] np.repeat
- [x] np.copy
- [x] np.diagonal
- [x] np.flatten
- [x] np.transpose
- [x] np.resize
- [x] np.ravel
- [x] np.compress
- [x] np.reshape
- [x] np.squeeze
- [x] np.swapaxes

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/tests/syft/core/tensor/smpc/smpc_tensor_test.py [hunk 1]
```diff
@@ -132,3 +132,186 @@ def test_mpc_forward_methods(method_str: str, kwargs: TypeDict[str, Any]) -> Non
...
+    )
+
+    res = mpc_tensor.partition(5).reconstruct()
+    exp_res = value.partition(5)
+
+    assert (res == exp_res).all()
+
+
+@pytest.mark.xfail
+def test_resize() -> None:
```
Comment (rasswanth-s) [inline packages/syft/tests/syft/core/tensor/smpc/smpc_tensor_test.py:230]: `ReSize`  tests fail due to refcheck in python, [Link](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.resize.html) ,will have to take additional look at it.

File: packages/syft/tests/syft/core/tensor/smpc/smpc_tensor_test.py [hunk 2]
```diff
@@ -132,3 +132,186 @@ def test_mpc_forward_methods(method_str: str, kwargs: TypeDict[str, Any]) -> Non
...
+    )
+
+    res = mpc_tensor.transpose().reconstruct()
+    exp_res = value.transpose()
+
+    assert (res == exp_res).all()
+
+
+@pytest.mark.xfail
+def test_partition() -> None:
```
Comment (rasswanth-s) [inline packages/syft/tests/syft/core/tensor/smpc/smpc_tensor_test.py]: Removed partition as it requires private compare

**Review by iamtrask — APPROVED** [review]

**Comment by iamtrask** [comment]
Well done @rasswanth-s! You made fast work of that!

File: packages/syft/tests/syft/core/tensor/smpc/smpc_tensor_test.py (hunk 1 above)
Comment (rasswanth-s) [reply packages/syft/tests/syft/core/tensor/smpc/smpc_tensor_test.py:213]: Modified to copy

**Pull request merged**